.build/images/dietpi-imager

```diff
@@ -493,7 +494,39 @@
 			G_EXEC partprobe "$FP_SOURCE"
 			G_EXEC partx -u "$FP_SOURCE"
 		fi
-
+		[[ -z ${ADD_DOS_PART+x} ]] && ADD_DOS_PART="0"
+		if [[ $ADD_DOS_PART == 1 ]]
+		then
+			G_DIETPI-NOTIFY 2 'Adding a 4MB FAT partition to the DOS partition table'
+			# figure out where the new partition starts
+			# WARNING: this assumes that the partitions in the table are in order (which we do in other places as well)
+			if [[ $SOURCE_TYPE == 'Image' ]]
+			then
+				G_DIETPI-NOTIFY 2 "working on a filesystem image, so adding 4MB to $FP_SOURCE_IMG"
+				G_EXEC_OUTPUT=1 G_EXEC dd if=/dev/zero bs=4M count=1 >> "$FP_SOURCE_IMG"
+			fi
+			local root_part=$(sfdisk -qlo DEVICE "$FP_SOURCE" | tail -1)
+			local np_start=$(($(sfdisk -qlo END "$FP_SOURCE" | tail -1) + 1))
+			local type='EBD0A0A2-B9E5-4433-87C0-68B6B72699C7'
+			[[ $PART_TABLE_TYPE == 'dos' ]] && type='c'
```

Comment:
Did you test whether the `+1` is needed with `sfdisk`?

It should not be needed to add a hardcoded GUID here, is it?
```suggestion
			local np_start=$(($(sfdisk -qlo END "$FP_SOURCE" | tail -1) + 1))
			local type=
			[[ $PART_TABLE_TYPE == 'dos' ]] && type=',type=c'
```
And then `echo "start=$np_start,size=8192$type"` below?

Reply:
> Did you test whether the `+1` is needed with `sfdisk`?

Yes. If you give it the end sector of the previous position it will error it with "sector is already used"

> It should not be needed to add a hardcoded GUID here, is it?

I believe it is. You need to mark this as a Windows partition so it gets mounted when plugged in on macOS or Windows.
And that crazy GUID is just the market for a Windows data partition.
Of course I could be reading the documentation wrong. 

> ```suggestion
> 			local np_start=$(($(sfdisk -qlo END "$FP_SOURCE" | tail -1) + 1))
> 			local type=
> 			[[ $PART_TABLE_TYPE == 'dos' ]] && type=',type=c'
> ```
> And then `echo "start=$np_start,size=8192$type"` below?

So what kind of position would this get you on gpt? My reading of the docs seemed to imply that by default you'd get a Linux partition. Which Windows or MacOS won't touch  regardless of the filesystem on it. But that's easy to try once I had breakfast 😁

Reply:
Ah you are right, I mixed up the UUID with the GUID somehow:
- https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_type_GUIDs
- https://en.wikipedia.org/wiki/Microsoft_basic_data_partition